commit/merkleroot/rmn/peerclient.go

```diff
@@ -3,21 +3,42 @@
 package rmn
 
 import (
-	"errors"
+	"context"
+	"crypto/sha256"
+	"encoding/binary"
+	"fmt"
+	"sync"
 
-	mapset "github.com/deckarep/golang-set/v2"
-	ragep2ptypes "github.com/smartcontractkit/libocr/ragep2p/types"
+	"github.com/smartcontractkit/libocr/commontypes"
+	"github.com/smartcontractkit/libocr/networking"
+	ocr2types "github.com/smartcontractkit/libocr/offchainreporting2plus/types"
+	"github.com/smartcontractkit/libocr/ragep2p"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	cciptypes "github.com/smartcontractkit/chainlink-common/pkg/types/ccipocr3"
 
 	rmntypes "github.com/smartcontractkit/chainlink-ccip/commit/merkleroot/rmn/types"
 )
 
-var (
-	ErrRMNNodeNotFound = errors.New("rmn node not found")
-)
+var ErrNoConn = fmt.Errorf("no connection, please call InitConnection before further interraction")
 
 // PeerClient performs low-level communication with RMN peers.
 type PeerClient interface {
-	Send(rmnNodeID rmntypes.NodeID, request []byte) error
+	// InitConnection initializes the connection to the peer group endpoint and must be called before
+	// further PeerClient interaction.
+	InitConnection(
+		ctx context.Context,
+		commitConfigDigest cciptypes.Bytes32,
+		rmnHomeConfigDigest cciptypes.Bytes32,
+		peerIDs []string, // union of oraclePeerIDs and rmnNodePeerIDs
```

Comment:
Explanation needed for why the union is required and not just the RMN node peer IDs. Is it something to do with peer discovery or something else?